File: app/src/main/java/ch/bailu/aat/views/preferences/AddOverlayDialog.java
package ch.bailu.aat.views.preferences;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;

import ch.bailu.aat.preferences.SolidOverlayFileList;
import ch.bailu.util_java.foc.Foc;

public class AddOverlayDialog extends AbsSolidDialog implements  DialogInterface.OnClickListener{

    private final SolidOverlayFileList slist;
    private final Foc file;


    public AddOverlayDialog (SolidOverlayFileList l, Foc f) {
        slist=l;
        file=f;

        final AlertDialog.Builder dialog = createDefaultDialog(l);
        dialog.setItems(slist.getStringArray(), this);
        dialog.create();
        dialog.show();
    }


    public AddOverlayDialog(Context context, Foc f) {
        this(new SolidOverlayFileList(context), f);
    }

    @Override
    public void onClick(DialogInterface dialog, int i) {
        slist.get(i).setValueFromFile(file);
        dialog.dismiss();
    }



}

File: aat-android/src/main/kotlin/ch/bailu/aat/views/preferences/dialog/AddOverlayDialog.kt
package ch.bailu.aat.views.preferences.dialog

import android.content.Context
import android.content.DialogInterface
import ch.bailu.aat_lib.app.AppContext
import ch.bailu.aat_lib.preferences.map.overlay.SolidCustomOverlay
import ch.bailu.aat_lib.preferences.map.overlay.SolidOverlayList
import ch.bailu.foc.Foc

class AddOverlayDialog(
    context: Context,
    private val slist: SolidOverlayList<SolidCustomOverlay>,
    private val file: Foc
) : AbsSolidDialog(), DialogInterface.OnClickListener {
    init {
        val dialog = createDefaultDialog(context, slist)
        dialog.setItems(slist.getStringArray(), this)
        dialog.create()
        dialog.show()
    }

    constructor(appContext: AppContext, context: Context, file: Foc) : this (
        context, SolidOverlayList.createCustomOverlayList(appContext), file
    )

    override fun onClick(dialog: DialogInterface, index: Int) {
        slist[index].setValueFromFile(file)
        dialog.dismiss()
    }
}
